core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,106 @@
+import org.broadinstitute.dsde.workbench.google2.{Location, MachineTypeName, NetworkName, SubnetworkName}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+/** Google Container Cluster statuses
+ *  see: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.Status
+ */
+
+case class KubernetesCluster(id: KubernetesClusterLeoId,
+                             googleProject: GoogleProject,
+                             clusterName: KubernetesClusterName,
+                             location: Location,
+                             status: KubernetesClusterStatus,
+                             serviceAccountInfo: WorkbenchEmail,
+                             samResourceId: KubernetesClusterSamResource,
+                             auditInfo: AuditInfo,
+                             asyncFields: KubernetesClusterAsyncFields,
+                             namespaces: Set[KubernetesNamespaceName],
+                             labels: LabelMap,
+                             nodepools: Set[Nodepool]
+                            )
+
+object KubernetesCluster {
+  implicit class EnrichedKubernetesCluster(cluster: KubernetesCluster) {
+    def getGKEClusterId: KubernetesClusterId = KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
+  }
+}
+
+case class SaveKubernetesCluster(googleProject: GoogleProject,
+                                 clusterName: KubernetesClusterName,
+                                 location: Location,
+                                 status: KubernetesClusterStatus,
+                                 serviceAccountInfo: WorkbenchEmail,
+                                 samResourceId: KubernetesClusterSamResource,
+                                 auditInfo: AuditInfo,
+                                 labels: LabelMap,
+                                 initialNodepool: Nodepool) //the clusterId specified here isn't used, and will be replaced by the id of cluster saved beforehand
+
+case class KubernetesClusterSamResource(resourceId: String)
+
+case class KubernetesClusterAsyncFields(apiServerIp: Option[KubernetesApiServerIp],
+                                        networkInfo: Option[NetworkFields]
+                                       )
+
+case class NetworkFields(networkName: NetworkName,
+                         subNetworkName: SubnetworkName,
+                         subNetworkIpRange: IpRange)
+
+
+//TODO: possibly consolidate Kubernetes status when monitoring is implemented
+sealed trait KubernetesClusterStatus extends EnumEntry with Product with Serializable
+object KubernetesClusterStatus extends Enum[KubernetesClusterStatus] {
+  val values = findValues
+
+  // NOTE: Remember to update the definition of this enum in Swagger when you add new ones
+  case object Status_Unspecified extends KubernetesClusterStatus
```

Comment:
Should ^^ just be `Unspecified`? Same below for `NodepoolStatus`.

Reply:
These words map directly (underscores and all for the below) to the google container statuses. The comment linking to the documentation got move as I wrote more code, I moved it to be next to these